src/openforms/submissions/tests/test_models.py

```diff
@@ -584,6 +586,22 @@ def test_clear_execution_state_without_execution_state(self):
         submission.clear_execution_state()
 
 
+class TemporaryFileUploadTests(TestCase):
+    def test_legacy_check_constraint(self):
+        with self.assertRaises(IntegrityError):
+            TemporaryFileUploadFactory.create(
+                submission=None,
+                legacy=False,
+            )
+
+    def test_non_legacy_check_constraint(self):
+        with self.assertRaises(IntegrityError):
+            TemporaryFileUploadFactory.create(
+                submission=SubmissionFactory.create(),
+                legacy=True,
+            )
```

Comment:
I find testing constraints on models with factories a bit tricky, I don't know what's happening behind the scenes and which implicit things are set up. I think I'd rather test model classes directly in those cases.

Reply:
`TemporaryFileUpload` requires some annoying fields (file fields)..